src/realm/object-store/sync/app_utils.cpp

```diff
@@ -77,7 +77,7 @@ util::Optional<AppError> AppUtils::check_for_errors(const Response& response)
                         response.http_status_code);
     }
 
-    return {};
+    return AppError(make_error_code(ServiceErrorCode::unknown), response.body.value_or("unexpected error"));
```

Comment:
I believe the intention of this method is to return an empty Optional if there are no errors discovered in the response. Won't this change make all responses return an error?

Reply:
Yes, you are right. this isn't the way for the fix.